From 3269b4ab250f9405254cd9aa858f418f964e6f80 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 7 Mar 2014 13:49:52 +0000
Subject: [PATCH 057/118] java: Fix bogus construction of all RStructList
 returned values (RHBZ#1073906).

Thanks Maarten on IRC for spotting the problem.

(cherry picked from commit d00840d727e4040a7958ecc613599d6ec2851e53)
---
 generator/java.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/generator/java.ml b/generator/java.ml
index 6ff6313..73ca3df 100644
--- a/generator/java.ml
+++ b/generator/java.ml
@@ -1188,7 +1188,7 @@ and generate_java_struct_list_return typ jtyp cols =
         pr "    fl = (*env)->GetFieldID (env, cl, \"%s\", \"C\");\n" name;
         pr "    (*env)->SetLongField (env, jfl, fl, r->val[i].%s);\n" name;
   ) cols;
-  pr "    (*env)->SetObjectArrayElement (env, jfl, i, jfl);\n";
+  pr "    (*env)->SetObjectArrayElement (env, jr, i, jfl);\n";
   pr "  }\n";
   pr "  guestfs_free_%s_list (r);\n" typ;
   pr "  return jr;\n"
-- 
2.7.4

